sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala

```diff
@@ -991,6 +991,13 @@ abstract class CatalogTestUtils {
 
   def newUriForDatabase(): URI = new URI(Utils.createTempDir().toURI.toString.stripSuffix("/"))
 
+  def newUriForPartition(parts: Seq[String]): URI = {
+    val path = parts.foldLeft(Utils.createTempDir()) {
+      case (parent, part) => new java.io.File(parent, part)
+    }
+    new URI(path.toURI.toString.stripSuffix("/"))
+  }
```

Comment:
no biggie but I would do:
```suggestion
  def newUriForPartition(parts: Seq[String]): URI = {
    val path = parts.foldLeft(Utils.createTempDir())(new java.io.File(_, _))
    new URI(path.toURI.toString.stripSuffix("/"))
  }
```